src/averageAges.js

```diff
@@ -37,7 +41,17 @@ function calculateMenAverageAge(people, century) {
  * @return {number}
  */
 function calculateWomenAverageAge(people, withChildren) {
-  // write code here
+  const filteredWomens = withChildren
+    ? people.filter((person) => {
+      return people.some(child => child.mother === person.name)
+        && person.sex === 'f';
+    })
```

Comment:
avoid return, where it's possible
```suggestion
    ? people.filter((person) => (
        people.some(child => child.mother === person.name) 
        && person.sex === 'f';
      ))
```